### PR TITLE
Trello 636: use Document entity without creating DocketRecords

### PR DESCRIPTION
**Pull request by rachaelparris**

### CODE REVIEW

File: shared/src/business/entities/cases/Case.js [hunk 1]
```diff
@@ -920,8 +920,19 @@ Case.prototype.addDocument = function (document, { applicationContext }) {
...
+      documentEntity,
+    });
+
+    if (updateIndex) {
+      documentEntity.index = this.generateNextDocketRecordIndex();
+    }
+  }
+
+  this.documents = [...this.documents, documentEntity];
 };
```
Comment (matthopson) [inline shared/src/business/entities/cases/Case.js:936]: Worth noting that this doing some docket record things in certain cases despite being called "withoutDocketRecord." This is necessary as an intermediate step and will be further refactored.

File: shared/src/business/entities/cases/Case.js [hunk 2]
```diff
@@ -1119,7 +1130,10 @@ Case.prototype.setRequestForTrialDocketRecord = function (
  * @returns {number} the next docket record index
  */
 Case.prototype.generateNextDocketRecordIndex = function () {
-  const recordsWithIndex = [...this.docketRecord]
+  const recordsWithIndex = [
+    ...this.docketRecord,
+    ...this.documents.filter(d => d.isOnDocketRecord),
+  ]
```
Comment (matthopson) [inline shared/src/business/entities/cases/Case.js:1136]: Splat references to both `docketRecord` and `documents` will also be refactored as `docketRecord` will go away completely.

File: shared/src/business/entities/Document.js [hunk 3]
```diff
@@ -122,6 +122,18 @@ function Document(rawDocument, { applicationContext, filtered = false }) {
...
+    rawDocument.description ||
+    rawDocument.documentTitle ||
+    rawDocument.documentType;
+  this.isStricken = rawDocument.isStricken || false;
+  this.strickenAt = rawDocument.strickenAt;
+  this.strickenBy = rawDocument.strickenBy;
+  this.strickenByUserId = rawDocument.strickenByUserId;
+  this.index = rawDocument.index;
+  this.servedPartiesCode = rawDocument.servedPartiesCode;
+  this.isOnDocketRecord = rawDocument.isOnDocketRecord || false;
```
Comment (matthopson) [inline shared/src/business/entities/Document.js:136]: While `docketRecord` properties are coming over to `document`, `isOnDocketRecord` is the flag that determines if an "entry" should exist for the document (and will likely serve in conditionally validating docket record related fields on the document)

File: shared/src/business/entities/Document.js [hunk 4]
```diff
@@ -653,4 +666,27 @@ Document.getFormattedType = function (documentType) {
...
+ * sets the number of pages for the docket entry
+ *
+ * @param {Number} numberOfPages the number of pages
+ */
+Document.prototype.setNumberOfPages = function (numberOfPages) {
+  this.numberOfPages = numberOfPages;
+};
+
+/**
+ * strikes this docket record
```
Comment (sutt0n) [inline shared/src/business/entities/Document.js:679]: "document" -- wonder if this should be conditional regarding `isOnDocketRecord`?

File: shared/src/business/entities/cases/PublicCase.js [hunk 5]
```diff
@@ -117,9 +117,10 @@ const isPrivateDocument = function (document, docketRecord) {
   const isCourtIssuedDocument = COURT_ISSUED_DOCUMENT_TYPES.includes(
     document.documentType,
   );
-  const isDocumentOnDocketRecord = docketRecord.find(
-    docketEntry => docketEntry.documentId === document.documentId,
-  );
+  const isDocumentOnDocketRecord =
```
Comment (sutt0n) [inline shared/src/business/entities/cases/PublicCase.js:120]: iirc, after migrations, this will just be `document.isOnDocketRecord`

Comment (rachaelparris) [reply shared/src/business/entities/cases/PublicCase.js:120]: Yes it will!

File: shared/src/business/entities/Document.js (hunk 4 above)
Comment (rachaelparris) [reply shared/src/business/entities/Document.js:679]: Would you be ok with adding this fix to another PR? I can add to the Trello task list.

**Review by sutt0n — APPROVED** [review]

**Review by codyseibert — APPROVED** [review]

**Pull request merged**